frontend/app/scripts/serve.js

```diff
@@ -109,7 +109,7 @@ const setupPreloadPackageWatcher = ({ ws }) =>
   try {
     const viteDevServer = await createServer({
       ...sharedConfig,
-      mode,
+      mode: process.env.CI && process.env.VITE_TEST ? 'production' : mode,
```

Comment:
can you explain about this part?

Reply:
So what I noticed in the failures on the CI was that sometimes vite would cause a page to reload in the mid of a test running. The reload doesn't seem to be due to HMR, we have that disabled when running tests on CI. 

I was looking at GitHub issues and one suggestion was to run vite in `production` mode to avoid these reloads.

So the idea is that if we are running tests on CI we don't want HMR or vite reloading, since this can cause test failures. For development mode we don't need that because it would cause problems when fixing/debugging tests.